mathesar/rpc/databases/configured.py

```diff
@@ -50,3 +50,17 @@ def list_(*, server_id: int = None, **kwargs) -> list[ConfiguredDatabaseInfo]:
         database_qs = Database.objects.all()
 
     return [ConfiguredDatabaseInfo.from_model(db_model) for db_model in database_qs]
+
+
+@rpc_method(name="databases.configured.disconnect")
+@http_basic_auth_login_required
+@handle_rpc_exceptions
+def disconnect(*, database_id: int, **kwargs) -> None:
+    """
+    Disconnect a configured database.
+
+    Args:
+        database_id: The Django id of the database.
+    """
+    database_qs = Database.objects.get(id=database_id)
+    database_qs.delete()
```

Comment:
We need to discuss on how to remove the Mathesar schemas. This requires a broader discussion on how we'll retain/request the user for the role & password for the role that owns the schemas.

We can get back to this before beta.